### ISSUE TITLE
generic.dockerfile.correctness.copy-from-own-alias.copy-from-own-alias matches even after switching images

### ISSUE THREAD

**Issue by malexmave** [returntocorp/sgrep-rules#1597]
**Describe the bug**
The rule [`generic.dockerfile.correctness.copy-from-own-alias.copy-from-own-alias`](https://semgrep.dev/r?q=generic.dockerfile.correctness.copy-from-own-alias.copy-from-own-alias) matches the following code:

```Dockerfile
FROM gradle:jdk11 as cache
RUN stuff

FROM other:1.0 as builder
COPY --from=cache /cache /home/gradle/.gradle

FROM image:3 as deploy
RUN stuff
```

[Reproduce in playground](https://semgrep.dev/s/vqEY)

**Expected behavior**
It should not match, because the described error (copying by alias from the currently active image) is not present. Again, I tried to fix it but could not make it work on short notice.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Comment by minusworld** [comment]
Blocked on returntocorp/semgrep#4327

**Comment by malexmave** [comment]
Bump to keep this open, as it is currently blocked on a different issue, but not necessarily stale.
